server/ot_rpc.go

```diff
@@ -131,22 +129,21 @@ func runTranscoder(n *core.LivepeerNode, orchAddr string, capacity int) error {
 }
 
 func runTranscode(n *core.LivepeerNode, orchAddr string, httpc *http.Client, notify *net.NotifySegment) {
-	var err error
-	var profiles []ffmpeg.VideoProfile
-	if len(notify.FullProfiles) > 0 {
-		profiles, err = makeFfmpegVideoProfiles(notify.FullProfiles)
-	} else if len(notify.Profiles) > 0 {
-		profiles, err = common.TxDataToVideoProfile(hex.EncodeToString(notify.Profiles))
-	}
-	if err != nil {
-		glog.Error("Unable to deserialize profiles ", err)
-	}
 
 	glog.Infof("Transcoding taskId=%d url=%s", notify.TaskId, notify.Url)
 	var contentType string
 	var body bytes.Buffer
 
-	tData, err := n.Transcoder.Transcode(notify.Job, notify.Url, profiles)
+	md, err := coreSegMetadata(notify.SegData)
+	if err != nil {
+		glog.Error("Unable to parse segData err=", err)
+		md = &core.SegTranscodingMetadata{} // avoid crash.
+		// TODO short-circuit error handling
```

Comment:
Perhaps we can short-circuit error handling by doing:

```golang
if err != nil {
    glog.Error("Unable to parse segData err=", err)
}
var tData *core.TranscodeData
if err == nil {
    tData, err = n.Transcoder.Transcode(md)
    ...
    if err == nil && len(tData.Segments) != len(profiles) {
        ...
    }
}
if err != nil {
    // Write error to body
}
```

Reply:
Yeah it was the large reflow of code I was hoping to avoid, plus additional work to test the error code etc. Error handling is generally broken at this point, though; this isn't an isolated case. One of the larger tech-debty things in the O/T protocol.

Reply:
Yeah the suggestion above was the least non-invasive change that came to mind, but definitely hear you that the structure could be cleaner/easier to test.

I'm fine with not going into a rabbit hole for this PR to fix all error handling in the O/T protocol. Created #1518 to track it - the description is pretty brief so feel free to add any relevant details there.

Reply:
Thanks for making the ticket. Added a reference to the ticket in the code - 89a3acfe2b7155b203e0945682bb2b98f9e3192c